test/gtest/ucs/test_async.cc

```diff
@@ -599,10 +599,6 @@ UCS_TEST_P(test_async, warn_block) {
     for (int i = 0; i < warn_count; ++i) {
         UCS_TEST_MESSAGE << "< " << m_warnings[i] << " >";
     }
-
-    if (GetParam() != UCS_ASYNC_MODE_POLL) {
```

Comment:
when you will handle return value from `ucs_spinlock_destroy`, print error using `ucs_warn`:
```
ucs_warn("ucs_spinlock_destroy() failed (%d)", status);
```
So, the test should be passed w/o any changes

@yosefe
btw, users (like UCM) may want to use `ucs_status_string()` but this is only declared in its code

Reply:
this test is supposed to count other type of warning, not spinlock destroy..

Reply:
if I got it correctly:
the test does `init() /* inside local_event ctor */` -> `block()` -> `destroy() /* inside local_event dtor */`
so, it expects warning from `local_event`'s dtor

Reply:
Should I change like the following? `1` -> `0`?
```C
if (GetParam() != UCS_ASYNC_MODE_POLL) {
    EXPECT_GE(warn_count, 0);
}
```

Reply:
> Should I change like the following? `1` -> `0`?

I think, it would be better to fix it by adding ucs_warn/ucm_warn for all places where we call `ucs_spinlock_destroy`
it should fix the problem

Reply:
@dmitrygx Thanks! Oh, I see.